cmake/CMakeLists.txt

```diff
@@ -146,6 +146,7 @@ option(onnxruntime_USE_MPI "Build with MPI support" OFF)
 
 # build WebAssembly
 option(onnxruntime_BUILD_WEBASSEMBLY "Enable this option to create WebAssembly byte codes" OFF)
+option(onnxruntime_ENABLE_WEBASSEMBLY_THREADS "Enable this option to create WebAssembly byte codes with multi-threads support" OFF)
```

Comment:
considering that we will support SIMD in future, it is better to have multiple targets for WASM.

`onnxruntime_BUILD_WEBASSEMBLY` as the ON/OFF switch for all WASM targets.
by default, WASM build target "onnxruntime_wasm"
`onnxruntime_ENABLE_WEBASSEMBLY_THREADS` add a new target "onnxruntime_wasm_threads"
`onnxruntime_ENABLE_WEBASSEMBLY_SIMD` add a new target "onnxruntime_wasm_simd"
and if both later 2 flags are enabled add a new target "onnxruntime_wasm_threads_simd"

we can define a set of C macros in code to support multiple targets.